test/TestAppUtils/VisualTreeDumper.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+
+namespace MUXControls.TestAppUtils
+{
+    public class VisualTreeDumper
+    {
+        public interface IFilter
+        {
+            bool ShouldLogElement(string elementName);
+            bool ShouldLogProperty(string propertyName);
+            bool IsKnownProperty(string propertyName, string value);
+        }
+
+        public interface IPropertyValueTranslator
+        {
+            string PropertyValueToString(string propertyName, Object value);
+        }
+
+        interface IVisitor
+        {
+            bool ShouldVisitNode(DependencyObject node);
+            void BeginVisitNode(DependencyObject node);
+            void EndVisitNode(DependencyObject node);
+
+            bool ShouldVisitPropertiesForNode(DependencyObject node);
+            bool ShouldVisitProperty(PropertyInfo propertyInfo);
+            void VisitProperty(String propertyName, Object value);
+        }
+
+        class Visitor : IVisitor
+        {
+            private StringBuilder _sb;
+            private int _indent;
+            private IFilter _filter;
+            private IPropertyValueTranslator _translator;
+            public Visitor(IFilter filter, IPropertyValueTranslator translator)
+            {
+                _sb = new StringBuilder();
+                _sb.AppendLine("<?xml version=\"1.0\" encoding=\"utf-8\" ?>");
```

Comment:
Use XmlWriter instead of doing this manually

Reply:
We don't use xml anymore